src/auditctl.c

```diff
@@ -1391,6 +1391,11 @@ static int fileopt(const char *file)
 		}
 		i = 0;
 		fields = malloc(nf * sizeof(char *));
+		if (fields == NULL) {
+			audit_msg(LOG_ERR, "Out of memory. Check %s file, %d line", __FILE__, __LINE__);
+			return 1;
```

Comment:
I would suggest to close the file descriptor before leaving the function, via `fclose(f);`